h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -387,54 +366,38 @@ public boolean hasNext(Model previousModel) {
           return false;
         }
 
-        @Override public void reset() {
-          _currentHyperparamIndices = null;
-        }
-
-        @Override
-        public double time_remaining_secs() { return Double.MAX_VALUE; }
-
-        @Override
-        public double max_runtime_secs() { return Double.MAX_VALUE; }
-
-        public int max_models() { return _maxHyperSpaceSize > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int)_maxHyperSpaceSize; }
-
         @Override
-        public void modelFailed(Model failedModel) {
-          // nada
+        public void onModelFailure(Model failedModel, Consumer<Object[]> withFailedModelHyperParams) {
+          // FIXME: when using parallel grid search, there's no good reason to think that the current hyperparam indices where the ones used for the failed model
+          withFailedModelHyperParams.accept(hypers(_currentHyperparamIndices));
```

Comment:
here, suggesting to keep a map of {model_params_checksum -> hyperparam_indices} instead of `_currentHyperparamIndices`, the latter being relevant only for grids run sequentially.